app/(storyblok)/[...slug]/page.tsx

```diff
@@ -45,29 +27,61 @@ async function getStoryData(params: { slug: string[] }) {
     cv: activeEnv === 'development' ? Date.now() : undefined,
     resolve_relations: resolveRelations,
   };
-
   const storyblokApi: StoryblokClient = getStoryblokApi();
-  const story = await storyblokApi?.get(`cdn/stories/${slug}`, sbParams);
 
+  try {
+  const story = await storyblokApi?.get(`cdn/stories/${slug}`, sbParams);
   return story;
+  } catch (error) {
+    notFound();
+  }
```

Comment:
the storyblock client throws an error when it can't find content. Catch it and 404 otherwise you'll get the error.tsx file.